src/lib/eth-calls.ts

```diff
@@ -60,4 +60,37 @@ async function setController(
   return await tx.wait();
 }
 
-export { registerTable, runSql, setController };
+async function getController(
+  this: Connection,
+  tableId: number
+): Promise<string> {
+  this.signer = this.signer ?? (await getSigner());
+
+  const contractAddress = this.options.contract;
+
+  const contract = TablelandTables__factory.connect(
+    contractAddress,
+    this.signer
+  );
+  return await contract.getController(tableId);
+}
+
+async function lockController(
+  this: Connection,
+  tableId: number
+): Promise<ContractReceipt> {
+  this.signer = this.signer ?? (await getSigner());
+  const caller = await this.signer.getAddress();
+
+  const contractAddress = this.options.contract;
+
+  const contract = TablelandTables__factory.connect(
+    contractAddress,
+    this.signer
+  );
+  const tx = await contract.lockController(caller, tableId);
+
+  return await tx.wait();
```

Comment:
Why diff from above. Just because this is an actual mutating call?

Reply:
Yeah, since getController is a `view` it returns a string. setController and lockController return transactions, and we've been returning the hash in those cases.